examples/elasticsearch-incremental/elasticsearch-incremental-blueprint.yaml

```diff
@@ -0,0 +1,542 @@
+          function dbg
+          {
+              if [[ $debug -eq 1 ]]
+              then 
+                echo "$(date "+%Y-%m-%d %H:%M:%S.%3N") - $0 - $*" >> /tmp/k10_debug.log
+              fi
+          }
+          function bp_echo
+          {
+            # use 
+            # kubectl logs -n kasten-io -l component=kanister --tail=10000 -f | ggrep -o -P '(?<=Elasticbps3echobegin).*(?=Elasticbps3echoend)' 
+            # to grab the logs in kanisters 
+            echo "Elasticbps3echobegin backup-action: $* Elasticbps3echoend"
+          }            
+          PROFILE_TYPE="{{ .Profile.Location.Type }}"
+          if [[ $PROFILE_TYPE != "s3Compliant" ]]
+          then 
+            dbg "Only s3Compliant profile are supported, exiting"
+            exit 1
+          fi
+          CLUSTER_UID=$(kubectl get ns default -o jsonpath='{.metadata.uid}')
+          NS_UID=$(kubectl get ns {{ .Object.metadata.namespace }} -o jsonpath='{.metadata.uid}')
+          ES_CLUSTER_UID={{ .Object.metadata.uid }}
+          REPO_PATH=k10/$CLUSTER_UID/elastic-search/$NS_UID/$ES_CLUSTER_UID
+          SNAPSHOT_NAME="snap_{{ toDate "2006-01-02T15:04:05.999999999Z07:00" .Time | date "2006-01-02t15:04:05z07:00" }}"
+          bp_echo "output the snapshot info $SNAPSHOT_NAME" 
+          kando output repoPath $REPO_PATH 
+          kando output snapshotName $SNAPSHOT_NAME 
+          kando output esClusterName {{ .Object.metadata.name }}
+
+          # setup the credentials on the keystore setting of each nodes 
+          {{- if .Profile.Credential.KeyPair }}
+          AWS_SECRET_KEY="{{ .Profile.Credential.KeyPair.Secret }}"
+          AWS_ACCESS_KEY="{{ .Profile.Credential.KeyPair.ID }}"
+          {{- else }}
+          AWS_SECRET_KEY="{{ .Profile.Credential.Secret.Data.aws_secret_access_key | toString }}"
+          AWS_ACCESS_KEY="{{ .Profile.Credential.Secret.Data.aws_access_key_id | toString }}"
+          {{- end }}
+          # No Support for more than one node set for the moment
+          NODE_SET_COUNT={{ (index .Object.spec.nodeSets 0).count }}
+          NODE_SET_COUNT=$((NODE_SET_COUNT-1))
+          NODE_SET_NAME={{ (index .Object.spec.nodeSets 0).name }}
+
+          for i in $(seq 0 $NODE_SET_COUNT)
+          do 
+              pod="{{ .Object.metadata.name }}-es-$NODE_SET_NAME-$i"
+              bp_echo "updating elasticsearch-keystore of $pod"
+              kubectl exec -it -n {{ .Object.metadata.namespace }} $pod -c elasticsearch -- bash -c "echo ${AWS_ACCESS_KEY} | /usr/share/elasticsearch/bin/elasticsearch-keystore add --stdin -f s3.client.default.access_key" 
+              kubectl exec -it -n {{ .Object.metadata.namespace }} $pod -c elasticsearch -- bash -c "echo ${AWS_SECRET_KEY} | /usr/share/elasticsearch/bin/elasticsearch-keystore add --stdin -f s3.client.default.secret_key"
+          done     
+          # make sure all output are grabbed by kanister pod 
+          sleep 4                                
```

Comment:
Hi @michaelcourcy 
do we need these `sleep` statements so that kanister controller can pick these logs? Are these logs not captures if we don't add these sleep statements.

Reply:
You're right this part could be removed